mirascope/core/base/_utils/_get_call_args_field_names_and_validate.py

```diff
@@ -0,0 +1,46 @@
+import inspect
+from collections.abc import Callable
+
+from pydantic import BaseModel
+from pydantic.fields import FieldInfo
+
+from mirascope.core.base.from_call_args import FromCallArgs
+
+
+def get_call_args_field_names_and_validate(
+    response_model: object, fn: Callable
+) -> set[str]:
+    if not (inspect.isclass(response_model) and issubclass(response_model, BaseModel)):
+        return set()
+    call_args_fields = _get_call_args_field_names(response_model)
+    _validate_call_args(fn, call_args_fields)
```

Comment:
i think this helper function is also unnecessary and could just live in this function:

```python
# validate call args
fn_args = get_fn_args(fn, args, kwargs)
if not call_args_fields.issubset(fn_args.keys()):
    raise ValueError("...")
return {name: fn_args[name] for name in call_args_fields}
```

note that part of this suggestion follows from my other comment around merging the mapping from earlier into this function

Reply:
The reason for not doing get_fn_args here is to validate the arguments on the function and FromCallArgs when defining the function.
If validation is performed together with `get_fn_args`, there will be no definition error unless a call is actually made and a response is received, thus delaying the detection of definition errors.

Therefore, merge the contents of existing helper functions

Reply:
but the validation still won't run until the user runs the function anyway right? it will still catch the issue before making the API call?

Reply:
We can get validation error when we define invalid call function.
```python
from typing import Annotated
from pydantic import BaseModel, Field
from mirascope.core import openai, FromCallArgs


class BookRecommendation(BaseModel):
    genre: Annotated[str, FromCallArgs()]
    title: str = Field(..., description="The title of the recommended book")
    author: str = Field(..., description="The author of the recommended book")


@openai.call("gpt-4o-mini", response_model=BookRecommendation)
def recommend_book(invalid_argument: str) -> str:
    return f"Recommend a {invalid_argument} book"
```

Output:
```bash
Traceback (most recent call last):
  File "/Users/koudai/Library/Application Support/JetBrains/PyCharm2024.2/scratches/scratch_49.py", line 12, in <module>
    @openai.call("gpt-4o-mini", response_model=BookRecommendation)
     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_extract.py", line 94, in decorator
    call_args_field_names = get_call_args_field_names_and_validate(
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/koudai/PycharmProjects/mirascope/mirascope/core/base/_utils/_get_call_args_field_names_and_validate.py", line 23, in get_call_args_field_names_and_validate
    raise ValueError(
ValueError: The function arguments do not contain all the fields marked with `FromCallArgs`. arguments=odict_keys(['invalid_argument']), call_args_fields={'genre'}

Process finished with exit code 1
```